src/infra/communication/connect.ts

```diff
@@ -0,0 +1,98 @@
+import assert from 'assert'
+import { Subject } from './subject'
+import { Observer } from './observer'
+
+type Resolver<T> = Record<string, (...args: any[]) => T | T[] | undefined>
+type Mutator = Record<string, (...args: any[]) => void>
```

Comment:
It would be useful if these types are in a separated file because we can use them on client side code.